packages/attestation-service/src/requestHandlers/attestation.ts

```diff
@@ -141,22 +150,51 @@ class AttestationRequestHandler {
     let attestation = await this.findOrValidateRequest()
 
     if (attestation && attestation.message) {
-      // Re-request existing attestation
+      // Re-request existing attestation. In this case, security code prefix is ignored (the message sent is the same as before)
       attestation = await rerequestAttestation(this.key, this.logger, this.sequelizeLogger)
     } else {
-      // New attestation: create text message, new delivery.
+      // New attestation: create new attestation code, new delivery.
       const attestationCode = await this.signAttestation()
       await this.validateAttestationCode(attestationCode)
+      const attestationCodeDeeplink = `celo://wallet/v/${toBase64(attestationCode)}`
 
-      const textMessage = createAttestationTextMessage(
-        attestationCode,
-        this.attestationRequest.smsRetrieverAppSig
-      )
+      // Determine if we're sending a security code, or the full deep link.
+      let messageBase, securityCode
+      if (this.attestationRequest.securityCodePrefix) {
+        if (this.attestationRequest.securityCodePrefix.length !== 1) {
+          throw new ErrorWithResponse('Invalid securityCodePrefix', 422)
+        }
+
+        // Client is requesting a security code SMS. Generate a challenge and just store the deeplink.
```

Comment:
hmm any reason we can't just sign the attestation when they are retrieving it during the `/get_attestation` request?  looks like all it needs is phoneNumber, salt, and account which we will have in that request as well.

just think we should persist as few codes as possible.

Reply:
nvm -- I guess we already persist the message today which is the signature. perhaps its not worth the added complexity for the `/get_attestation` method